src/main/java/com/sngular/kloadgen/extractor/impl/SchemaExtractorImpl.java

```diff
@@ -55,20 +59,45 @@ public SchemaExtractorImpl(final AvroExtractor avroExtractor, final JsonExtracto
-    } else if (SchemaTypeEnum.PROTOBUF.name().equalsIgnoreCase(schema.schemaType())) {
-      final var protoFileElement = ((ProtobufSchema) schema).rawSchema();
-      final HashMap<String, TypeElement> nestedTypes = new HashMap<>();
-      protoFileElement.getTypes().forEach(field -> protoBufExtractor.processField(field, attributeList, protoFileElement.getImports(), false, nestedTypes));
-    } else {
-      throw new KLoadGenException(String.format("Schema type not supported %s", schema.schemaType()));
+
+    String registryName = properties.getProperty(SchemaRegistryKeyHelper.SCHEMA_REGISTRY_NAME);
+    if (Objects.nonNull(registryName) && registryName.equalsIgnoreCase(SchemaRegistryEnum.APICURIO.toString())) {
+      ApicurioParsedSchema apicurioParsedSchema = (ApicurioParsedSchema) schemaParsed;
+      Object schema = apicurioParsedSchema.getSchema();
+
+      schemaType = apicurioParsedSchema.getType();
+      if (SchemaTypeEnum.AVRO.name().equalsIgnoreCase(schemaType)) {
+        ((Schema) schema).getFields().forEach(field -> avroExtractor.processField(field, attributeList, true, false));
+      } else if (SchemaTypeEnum.JSON.name().equalsIgnoreCase(schemaType)) {
+        attributeList.addAll(jsonExtractor.processSchema(((io.apicurio.registry.serde.jsonschema.JsonSchema) schema).toJsonNode()));
+      } else if (SchemaTypeEnum.PROTOBUF.name().equalsIgnoreCase(schemaType)) {
+        final var protoFileElement = ((io.apicurio.registry.utils.protobuf.schema.ProtobufSchema) schema).getProtoFileElement();
+        protoFileElement.getTypes().forEach(field -> protoBufExtractor.processField(field, attributeList, protoFileElement.getImports(), false, null));
+      } else {
+        throw new KLoadGenException(String.format("Schema type not supported %s", apicurioParsedSchema.getType()));
+      }
+
+    } else if (Objects.nonNull(registryName) && registryName.equalsIgnoreCase(SchemaRegistryEnum.CONFLUENT.toString())) {
+      ParsedSchema confluentParsedSchema = (ParsedSchema) schemaParsed;
+
+      schemaType = confluentParsedSchema.schemaType();
+      if (SchemaTypeEnum.AVRO.name().equalsIgnoreCase(schemaType)) {
+        (((AvroSchema) confluentParsedSchema).rawSchema()).getFields().forEach(field -> avroExtractor.processField(field, attributeList, true, false));
+      } else if (SchemaTypeEnum.JSON.name().equalsIgnoreCase(schemaType)) {
+        attributeList.addAll(jsonExtractor.processSchema(((JsonSchema) confluentParsedSchema).toJsonNode()));
+      } else if (SchemaTypeEnum.PROTOBUF.name().equalsIgnoreCase(schemaType)) {
+        final var protoFileElement = ((ProtobufSchema) confluentParsedSchema).rawSchema();
+        protoFileElement.getTypes().forEach(field -> protoBufExtractor.processField(field, attributeList, protoFileElement.getImports(), false, null));
+      } else {
+        throw new KLoadGenException(String.format("Schema type not supported %s", schemaType));
+      }
```

Comment:
Maybe these two blocks of code could be extracte into two functions, the logic is a bit hard to read at a glance

Reply:
You are right. It is planned to refactor in branch #342